internal/provider/resource_eva_role.go

```diff
@@ -0,0 +1,270 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/mad-it/terraform-provider-eva/internal/eva"
+)
+
+type roleProviderType struct{}
+
+func (t roleProviderType) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		MarkdownDescription: "Eva role configuration.",
+
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				MarkdownDescription: "ID of the role.",
+				Computed:            true,
+				Type:                types.Int64Type,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					tfsdk.UseStateForUnknown(),
+				},
+			},
+			"name": {
+				MarkdownDescription: "Name of the role",
+				Optional:            true,
+				Type:                types.StringType,
+			},
+			"user_type": {
+				MarkdownDescription: "User type this role applies to",
+				Optional:            true,
+				Type:                types.Int64Type,
+			},
+			"code": {
+				MarkdownDescription: "A unique code to represent the role",
+				Optional:            true,
+				Type:                types.StringType,
+			},
+			"scoped_functionalities": {
+				MarkdownDescription: "list of scoped functionalities to be attached",
+				Required:            true,
+				Attributes: tfsdk.ListNestedAttributes(
+					map[string]tfsdk.Attribute{
+						"functionality": {
+							MarkdownDescription: "functionality identifier",
+							Required:            true,
+							Type:                types.StringType,
+						},
+						"scope": {
+							MarkdownDescription: "functionality scope",
+							Required:            true,
+							Type:                types.Int64Type,
+						},
+						"requires_elevation": {
+							MarkdownDescription: "whether functionality requires elevation or not",
+							Required:            true,
+							Type:                types.BoolType,
+						},
```

Comment:
![image](https://user-images.githubusercontent.com/1469047/157660492-11a6f7b4-7e56-4dda-8746-7e42085c19e9.png)
Can we set default to false for this one? That way we match current importer functionality and dont need to specify it for each role.

Reply:
Good catch, I can implement a modifier for the case value is not present in the plan configuration.